src/reconcile/rec_write.c

```diff
@@ -3090,10 +3099,18 @@ __rec_write_wrapup(WT_SESSION_IMPL *session, WTI_RECONCILE *r)
             break;
         }
 
-        WT_RET(__wt_ref_block_free(session, ref,
-          page->disagg_info != NULL &&
-            page->disagg_info->block_meta.page_id != WT_BLOCK_INVALID_PAGE_ID &&
-            r->multi_next != 1));
+        /*
+         * Free the disaggregated block if reconciliation results in zero pages, multiple pages, or
+         * a single empty page.
+         */
+        if (disagg_page_is_valid)
+            /*
+             * r->multi == NULL implies r->multi_next == 0; thus it is safe to access block_meta
+             * directly.
+             */
+            disagg_page_free_required =
+              (r->multi_next != 1 || r->multi->block_meta->page_id == WT_BLOCK_INVALID_PAGE_ID);
```

Comment:
Does the second part of the condition cover "or a single empty page" from the comment?